flink-connector-aws/flink-connector-sqs/src/main/java/org/apache/flink/connector/sqs/sink/SqsStateSerializer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.sqs.sink;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.connector.base.sink.writer.AsyncSinkWriterStateSerializer;
+
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequestEntry;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+/** SQS implementation {@link AsyncSinkWriterStateSerializer}. */
+@Internal
+public class SqsStateSerializer
+        extends AsyncSinkWriterStateSerializer<SendMessageBatchRequestEntry> {
+    @Override
+    protected void serializeRequestToStream(
+            final SendMessageBatchRequestEntry request, final DataOutputStream out)
+            throws IOException {
+        out.write(request.messageBody().getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    protected SendMessageBatchRequestEntry deserializeRequestFromStream(
+            final long requestSize, final DataInputStream in) throws IOException {
+        final byte[] requestData = new byte[(int) requestSize];
+        in.read(requestData);
+        return SendMessageBatchRequestEntry.builder()
+                .id(UUID.randomUUID().toString())
```

Comment:
This is a little bit weird that we generate a random UUID each time the stat is serialized. Flink does this transparently - it would be good to keep this in state + ensure the UUID is kept constant for a given request

Reply:
@hlteoh37 We need to generate this request ID at each batch request entry level, that's the SQS request contract format. We can have multiple batch request entry in each batch, hence generated here.

When you suggested generating it at request level, what do you mean by that? Any references for keeping in this state?

Reply:
No that makes sense, but the comment here is more around not re-generating the `UUID` every time we serialize / deserialize the request. Instead, can we include the UUID in the serialization and retrieve it when we deserialize it?

Reply:
Make sense, updated the code accordingly